src/Data/Nat/Properties.agda

```diff
@@ -968,25 +972,25 @@ n≢0∧m>1⇒m*n>1 m n rewrite *-comm m n = m≢0∧n>1⇒m*n>1 n m
 *-cancelˡ-≤ : ∀ o .{{_ : NonZero o}} → o * m ≤ o * n → m ≤ n
 *-cancelˡ-≤ {m} {n} o rewrite *-comm o m | *-comm o n = *-cancelʳ-≤ m n o
 
-*-mono-≤ : _*_ Preserves₂ _≤_ ⟶ _≤_ ⟶ _≤_
-*-mono-≤ z≤n       _   = z≤n
+*-mono-≤ : Monotonic₂ _≤_ _≤_ _≤_ _*_
+*-mono-≤ z≤n       u≤v = z≤n
 *-mono-≤ (s≤s m≤n) u≤v = +-mono-≤ u≤v (*-mono-≤ m≤n u≤v)
 
-*-monoˡ-≤ : ∀ n → (_* n) Preserves _≤_ ⟶ _≤_
-*-monoˡ-≤ n m≤o = *-mono-≤ m≤o (≤-refl {n})
+*-monoˡ-≤ : RightMonotonic _≤_ _≤_ _*_
+*-monoˡ-≤ = mono₂⇒monoʳ {≤₁ = _≤_} {≤₂ = _≤_} {≤₃ = _≤_} ≤-refl *-mono-≤
 
-*-monoʳ-≤ : ∀ n → (n *_) Preserves _≤_ ⟶ _≤_
-*-monoʳ-≤ n m≤o = *-mono-≤ (≤-refl {n}) m≤o
+*-monoʳ-≤ : LeftMonotonic _≤_ _≤_ _*_
+*-monoʳ-≤ = mono₂⇒monoˡ {≤₁ = _≤_} {≤₂ = _≤_} {≤₃ = _≤_} ≤-refl *-mono-≤
```

Comment:
This providing of implicits is annoying. Suggests that some of them should be explicit in the proof?

Reply:
Yes, I've been scratching my head over how it might be avoided... without a conclusion yet.

Reply:
So if you make all 3 underscores, which ones are yellow? I notice that the previous proof provided the `n` explicitly, might that work here as well?

Reply:
It seems that only `{≤₃ = _≤_}` is required here...

Reply:
Aaargh: there's a difference depending on whether we consider left or right, presumably because of the left-biased implementations. Will need to investigate further. Sigh.
```agda
*-monoˡ-≤ : RightMonotonic _≤_ _≤_ _*_
*-monoˡ-≤ = mono₂⇒monoʳ {≤₃ = _≤_} ≤-refl *-mono-≤

*-monoʳ-≤ : LeftMonotonic _≤_ _≤_ _*_
*-monoʳ-≤ = mono₂⇒monoˡ {≤₁ = _≤_} {≤₂ = _≤_} {≤₃ = _≤_} ≤-refl *-mono-≤
```